scripts/linux_installer.py

```diff
@@ -0,0 +1,92 @@
+        dns = stdout.removeprefix("dns=").strip()
+        if dns == "dnsmasq":
+            logging.info("NetworkManager is already configured")
+        elif dns == "":
+            # add dns=dnsmasq after [main] line
+            self.executeCommand("sudo sed -i 's/^\[main\]$/\[main\]\\ndns=dnsmasq/' /etc/NetworkManager/NetworkManager.conf", True)
+        else:
+            self.executeCommand("sudo sed -i 's/^dns=.*/dns=dnsmasq/g' /etc/NetworkManager/NetworkManager.conf", True)
+
+    def configureDNSMasq(self):
+        logging.info("Configuring DNSMasq...")
+
+        self.executeCommand("sudo mkdir -p /etc/NetworkManager/dnsmasq.d", True)
+        self.executeCommand("sudo bash -c 'echo 'address=/.massa/127.0.0.1' > /etc/NetworkManager/dnsmasq.d/massa.conf'", True)
+        self.executeCommand("sudo mv /etc/resolv.conf /etc/resolv.conf.backup_thyra_install", True)
+        self.executeCommand("sudo ln -s /run/NetworkManager/resolv.conf /etc/resolv.conf", True)
+        self.executeCommand("sudo systemctl restart NetworkManager", True)
+
+    def setupDNS(self):
+        stdout, _stderr = self.executeCommand("sudo lsof -i :53", True, allow_failure=True)
+        runningDNS = ""
+        if stdout:
+            runningDNS = stdout.splitlines()[1].split()
+            runningDNS = runningDNS[0]
+
+        if runningDNS == "":
+            logging.info("Installing dnsmasq...")
+            self.executeCommand("sudo apt-get install -y dnsmasq", True)
+            self.configureNetworkManager()
+            self.configureDNSMasq()
+        elif runningDNS == "dnsmasq":
+            logging.info("dnsmasq is already installed")
+            self.configureDNSMasq()
+        elif runningDNS == "systemd-r":
+            logging.warning("Your computer is using systemd-resolved for DNS. Thyra needs dnsmasq to redirect .massa domains to localhost.")
+            user_answer = input("Do you want to install dnsmasq and configure it to redirect .massa domains to localhost? [y/n] ")
+            user_answer = user_answer.lower()
+            if user_answer == "y" or user_answer == "yes":
+                self.configureNetworkManager()
+                self.configureDNSMasq()
+            else:
+                self.printErrorAndExit("Aborting installation.")
+        else:
+            logging.warning(f"Unsupported DNS application: {runningDNS}")
+
+    def generateCACertificate(self):
+        stdout, _stderr = self.executeCommand("firefox -v && echo \"firefox installed\"", True, allow_failure=True)
+        if stdout and "installed" in stdout:
+            self.executeCommand("sudo apt-get install -y libnss3-tools", True)
+
+        super().generateCACertificate()
+    
+    def _moveFile(self, file, destination):
+        self.executeCommand(f"sudo mv {file} {destination}", True)
+
+    def _deleteFile(self, file):
+        self.executeCommand(f"sudo rm {file}", True)
```

Comment:
I think here you can use the methods defined in base classe Installer

Reply:
No, the command needs sudo...